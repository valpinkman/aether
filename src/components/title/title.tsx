import styled from 'styled-components'
import { color, ColorProps, typography, TypographyProps } from 'styled-system'

type Props = ColorProps & TypographyProps

export const Title = styled.h1<Props>`
  font-family: Ailerons, sans-serif;
  ${color}
  ${typography}
`

export const Subtitle = styled.h2<Props>`
  ${color}
  ${typography}
`
