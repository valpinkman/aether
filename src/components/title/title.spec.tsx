import React from 'react'
import { shallow } from 'enzyme'
import { Title, Subtitle } from './title'

describe('components', () => {
  describe('<Title />', () => {
    it('should render', () => {
      const wrapper = shallow(<Title />)
      expect(wrapper).toMatchSnapshot()
    })

    it('should render correct text', () => {
      const wrapper = shallow(<Title>Hello World</Title>)

      expect(wrapper.prop('children')).toBe('Hello World')
      expect(wrapper).toMatchSnapshot()
    })

    it('should access color props', () => {
      const wrapper = shallow(<Title color="text.shade100" />)

      expect(wrapper.prop('color')).toBe('text.shade100')
      expect(wrapper).toMatchSnapshot()
    })

    it('should access typography props', () => {
      const wrapper = shallow(<Title fontSize={2} />)

      expect(wrapper.prop('fontSize')).toBe(2)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('<Subtitle />', () => {
    it('should render', () => {
      const wrapper = shallow(<Subtitle />)
      expect(wrapper).toMatchSnapshot()
    })

    it('should render correct text', () => {
      const wrapper = shallow(<Subtitle>Hello World</Subtitle>)

      expect(wrapper.prop('children')).toBe('Hello World')
      expect(wrapper).toMatchSnapshot()
    })

    it('should access color props', () => {
      const wrapper = shallow(<Subtitle color="text.shade100" />)

      expect(wrapper.prop('color')).toBe('text.shade100')
      expect(wrapper).toMatchSnapshot()
    })

    it('should access typography props', () => {
      const wrapper = shallow(<Subtitle fontSize={2} />)

      expect(wrapper.prop('fontSize')).toBe(2)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
