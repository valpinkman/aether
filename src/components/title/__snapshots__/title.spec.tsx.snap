// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <Subtitle /> should access color props 1`] = `
<StyledComponent
  color="text.shade100"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": false,
        "rules": Array [
          "
  ",
          [Function],
          "
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.h2",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "h2",
      "toString": [Function],
      "usesTheme": true,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`components <Subtitle /> should access typography props 1`] = `
<StyledComponent
  fontSize={2}
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": false,
        "rules": Array [
          "
  ",
          [Function],
          "
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.h2",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "h2",
      "toString": [Function],
      "usesTheme": true,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`components <Subtitle /> should render 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": false,
        "rules": Array [
          "
  ",
          [Function],
          "
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.h2",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "h2",
      "toString": [Function],
      "usesTheme": true,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`components <Subtitle /> should render correct text 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": false,
        "rules": Array [
          "
  ",
          [Function],
          "
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.h2",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "h2",
      "toString": [Function],
      "usesTheme": true,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  Hello World
</StyledComponent>
`;

exports[`components <Title /> should access color props 1`] = `
<StyledComponent
  color="text.shade100"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  font-family: Ailerons, sans-serif;
  ",
          [Function],
          "
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.h1",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "h1",
      "toString": [Function],
      "usesTheme": true,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`components <Title /> should access typography props 1`] = `
<StyledComponent
  fontSize={2}
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  font-family: Ailerons, sans-serif;
  ",
          [Function],
          "
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.h1",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "h1",
      "toString": [Function],
      "usesTheme": true,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`components <Title /> should render 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  font-family: Ailerons, sans-serif;
  ",
          [Function],
          "
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.h1",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "h1",
      "toString": [Function],
      "usesTheme": true,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`components <Title /> should render correct text 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  font-family: Ailerons, sans-serif;
  ",
          [Function],
          "
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.h1",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "h1",
      "toString": [Function],
      "usesTheme": true,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  Hello World
</StyledComponent>
`;
